cirq/protocols/resolve_parameters.py

```diff
@@ -63,8 +76,54 @@ def is_parameterized(val: Any) -> bool:
 
     if result is not NotImplemented:
         return result
-    else:
-        return False
+
+    return bool(parameter_names(val))
+
+
+def parameter_names(val: Any) -> AbstractSet[str]:
+    """Returns parameter names for this object.
+
+    Args:
+        val: Object for which to find the parameter names.
+        check_symbols: If true, fall back to calling parameter_symbols.
+
+    Returns:
+        A set of parameter names if the object is parameterized. It the object
+        does not implement the _parameter_names_ magic method or that method
+        returns NotImplemented, returns an empty set.
+    """
+    if isinstance(val, sympy.Basic):
+        return {symbol.name for symbol in val.free_symbols}
+    if isinstance(val, numbers.Number):
```

Comment:
I would move this up to be first as well.

Reply:
Done.